main.go

```diff
@@ -1,9 +1,482 @@
+	rmserr = Sx(s) / math.Sqrt(2*float64(len(s)))
+
+	return rmserr
+}
+
+func MEANerr(s []float64) float64 {
+
+	// compute the error on the mean of a given distribution
+
+	var meanerr float64
+	meanerr = Sx(s) / math.Sqrt(float64(len(s)))
+
+	return meanerr
+}
+
+func difference(s1, s2 []float64) ([]float64, []float64) {
+
+	// compute the difference between two slices, value by value
+
+	var diff_slice []float64
+	var abs_diff_slice []float64
+
+	for i,_ := range(s1) {
+		val := s2[i] - s1[i]
+		diff_slice = append(diff_slice, val)
+		abs_diff_slice = append(abs_diff_slice, math.Abs(val))
+	}
+
+	return diff_slice, abs_diff_slice
+}
+
+/*
+#####
+Theoretical mu
+#####
+*/
+
+func integral(z, n float64) float64 {
+
+	// compute the integral of the D_l function, based on the trapeze method
+	// integrate.Trapezoidal takes two slices (abscissa and ordinate) from which to compute the integral
+	// inputs : z = redshift, n = number of steps for the trapeze method
+
+	var inte float64
+	abs := make([]float64, 0)
+	ord := make([]float64, 0)
+
+	var i float64
+	for i = 0; i <= n; i++ {
+		val := z * i / n
+		abs = append(abs, val)
+	}
+
+	for j := 0; j <= int(n); j++ {
+		val := 1 / math.Sqrt((1+abs[j]*omgM)*(1+abs[j])*(1+abs[j])-(1-omgM)*(2+abs[j])*abs[j])
+		ord = append(ord, val)
+	}
+
+	inte = integrate.Trapezoidal(abs, ord)
+
+	return inte
+
+}
+
+func mu_th(z, n float64) float64 {
+
+	// compute the theoretical mu for each SN
+	// inputs : z = redshift, n = number of steps for the integral, to be passed to integral function
+
+	var mu_th float64
+	mu_th = 5 * math.Log10(((1+z)*c/(10*H))*integral(z, n))
+
+	return mu_th
+}
+
+func mu_th_slice(s []float64, n float64) []float64 {
+
+	// builds a slice containing the theoretical distances
+	// inputs : s = slice of the zcmb, n = number of steps for the integral, to be passed to the integral function through mu_th
+
+	diag_ord := make([]float64, 0)
+
+	for i := 0; i < len(s); i++ {
+		diag_ord = append(diag_ord, mu_th(s[i], n))
+	}
+
+	return diag_ord
+}
+
+/*
+######
+Experimental mu
+######
+*/
+
+func mu_exp_slice(m_stell, mb, stretch, colour []float64) []float64 {
+
+	// compute experimental mus and add them in an array
+	// inputs : mB = B band peak magnitude, stretch and colour = SALT2 parameters
+
+	slice := make([]float64, 0)
+	for i := 0; i < len(mb); i++ {
+		if m_stell[i] < 10 {
+			val := mb[i] - Mb + alpha*stretch[i] - beta*colour[i]
+			slice = append(slice, val)
+		} else {
+			val := mb[i] - Mb - delta_M + alpha*stretch[i] - beta*colour[i]
+			slice = append(slice, val)
+		}
+	}
+
+	return slice
+}
+
+func mu_exp_err(dmb, dstretch, dcolour []float64) []float64 {
+
+	// compute errors on experimental mus and add them in an array
+	// inputs : dmb, dstretch, dcolour = errors on mb, stretch and colour from JLA data
+
+	slice := make([]float64, 0)
+	for i := 0; i < len(dmb); i++ {
+		val := math.Sqrt(math.Pow(dmb[i], 2) + math.Pow(alpha*dstretch[i], 2) + math.Pow(beta*dcolour[i], 2))
```

Comment:
`math.Pow` isn't supposed to be used for such trivial powers (ie: it's not very fast because it needs to be able to handle floating point powers...)

better create 3 temporary values and multiply:

```go
slice := make([]float64, len(dmb))
for i := range dmb {
  ai := dmb[i]
  bi := alpha*dstretch[i]
  ci := beta*dcoulour[i]
  slice[i] = math.Sqrt(ai*ai + bi*bi + ci*ci)
}
return slice
```